runtime/interpreter/value.go

```diff
@@ -6668,6 +6779,28 @@ func (v *CompositeValue) ConformsToDynamicType(
 	return true
 }
 
+func (v *CompositeValue) IsStorable() bool {
+
+	// Composite value's of native/built-in types are not storable for now
+	if v.Location() == nil {
+		return false
+	}
+
+	// If this composite value has a field which is non-storable,
+	// then the composite value is not storable.
```

Comment:
We check for the existence of computed fields when checking for storability of a [CompositeType](https://github.com/onflow/cadence/blob/master/runtime/sema/type.go#L3172:L3174) at the checker. Do we need to do the same here?

Reply:
Oh, we shouldn't consider computed fields when determining storability, because computed fields are not stored, right?

Reply:
Idea was, at the time of decoding, we wouldn't know how to compute the field. 
We could still allow it I guess, but then we'll have to special case the types with computed fields to, and specifically construct a value of that type. e.g: if the type is `PublicKey`, use the `NewPublicKey` constructor, etc.

I think this has no effect for the moment, since only built-in types can have computed fields, and all built-in types are not storable anyway.

Reply:
Makes sense! I'll leave it as-is for now, as we don't readily have access to `hasComputedMembers` of the type here.

Still, I had a look at the type's storability function, and noticed it was checking the composite kind. I've added that in 85109787507fe559c51d8a74f7a72a078ef1eee8 to the value (see the comment why contracts are included)

Reply:
oh right, that's a good catch!

For the `hasComputedMembers`, instead maybe we could check the existence of the `ComputedFields` in the composite value. The `hasComputedMembers` is a de-facto for the same. I added that flag only because there is no separation between computed vs normal fields at the type level, like in the value.